src/gl/canvas2dRe.re

```diff
@@ -0,0 +1,76 @@
+type t;
+
+/* Canvas API, following https://simon.html5.org/dump/html5-canvas-cheat-sheet.html */
+
+external save : unit = "" [@@bs.send.pipe: t];
+external restore : unit = "" [@@bs.send.pipe: t];
+
+/* Transformation */
+external scale : float => float => unit = "" [@@bs.send.pipe: t];
+external rotate : float => unit = "" [@@bs.send.pipe: t];
+external translate : float => float => unit = "" [@@bs.send.pipe: t];
+external transform : m11::float => m12::float => m21::float => m22::float => dx::float => dy::float => unit = "" [@@bs.send.pipe: t];
+external setTransform : m11::float => m12::float => m21::float => m22::float => dx::float => dy::float => unit = "" [@@bs.send.pipe: t];
+
+/* Compositing */
+external globalAlpha : t => float => unit = "" [@@bs.set];
+external globalCompositeOperation : t => string => unit = "" [@@bs.set];
```

Comment:
I believe my latest commit has addressed this– it's a bit wordy but I added a module interface for each of the enum submodules. Now the compiler gives you an error if you pass a random string instead of one of the enum values.